presto-main/src/test/java/com/facebook/presto/operator/TestHashJoinOperator.java

```diff
@@ -283,41 +354,51 @@ public void testInnerJoinWithSpill(boolean probeHashEnabled, List<WhenSpill> whe
                 runDriver(executor, buildDriver);
             }
 
-            // process join
-            Iterator<Page> input = probePages.build().iterator();
-            List<Page> actualPages = new ArrayList<>();
-            actualPages.addAll(toPagesPartial(joinOperator, singletonIterator(input.next())));
+            ValuesOperatorFactory valuesOperatorFactory = new ValuesOperatorFactory(17, new PlanNodeId("values"), probePages.getTypes(), probePages.build());
+
+            PageBuffer pageBuffer = new PageBuffer(10);
+            PageBufferOperatorFactory pageBufferOperatorFactory = new PageBufferOperatorFactory(18, new PlanNodeId("pageBuffer"), pageBuffer);
+
+            Driver ljoDriver = new Driver(ljoDriverContext,
+                                          valuesOperatorFactory.createOperator(ljoDriverContext),
+                                          joinOperator,
+                                          pageBufferOperatorFactory.createOperator(ljoDriverContext));
+
+            while (!called.get()) { // process first row of first page of LookupJoinOperator
+                ljoDriver.getDriverContext().getYieldSignal().setWithDelay(10000, ljoDriver.getDriverContext().getYieldExecutor());
+                tryGetFutureValue(ljoDriver.process(), 10, TimeUnit.MILLISECONDS);
+                ljoDriver.getDriverContext().getYieldSignal().reset();
+                checkErrors(taskStateMachine);
+            }
 
             for (int i = 0; i < buildOperatorCount; i++) {
                 if (whenSpill.get(i) == WhenSpill.DURING_USAGE) {
-                    checkState(input.hasNext(), "spill requested DURING_USAGE, but input is kind of exhausted");
-                    triggerMemoryRevokingAndWait(buildSideSetup.buildOperators.get(i));
+                    triggerMemoryRevokingAndWait(buildSideSetup.buildOperators.get(i), taskStateMachine);
                 }
             }
 
-            actualPages.addAll(toPages(joinOperator, input));
+            // process remaining LookupJoinOperator pages
+            while (!ljoDriver.isFinished()) {
+                ljoDriver.getDriverContext().getYieldSignal().setWithDelay(10000, ljoDriver.getDriverContext().getYieldExecutor());
```

Comment:
`10000` doesn't disclose the time delay (unless you look at the code in IDE and have param hits or something).
What about extracting this to a method which name would convey meaning? Or would take `long, TimeUnit` (or `Duration`)?

Also, this is in nanos, so 10000 is rather small amount of time.

Reply:
Or maybe the whole block `set delay; process; reset yield;` could be extracted to a method?